src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp

```diff
@@ -522,6 +451,9 @@ class boss_twilight_halion : public CreatureScript
 
             void DamageTaken(Unit* attacker, uint32& damage) override
             {
+                if (!events.IsInPhase(PHASE_TWO) && !events.IsInPhase(PHASE_THREE))
+                    EnterCombat(attacker);
```

Comment:
this became needed because of the UNIT_FLAG_IN_COMBAT, the EnterCombat hook will never be called unless I explicitly do it.